apps/web/src/pages/templates/components/chat-editor/TemplateChatEditor.tsx

```diff
@@ -1,51 +1,76 @@
 import { ChannelTypeEnum } from '@novu/shared';
 import { Controller, useFormContext } from 'react-hook-form';
+import { Grid, Stack } from '@mantine/core';
+import { useState } from 'react';
+import { useTimeout } from '@mantine/hooks';
 
-import { Textarea } from '@novu/design-system';
-import { useEnvController, useHasActiveIntegrations, useVariablesManager } from '../../../../hooks';
-import { useStepFormErrors } from '../../hooks/useStepFormErrors';
+import { useHasActiveIntegrations, useVariablesManager } from '../../../../hooks';
 import { useStepFormPath } from '../../hooks/useStepFormPath';
 import { StepSettings } from '../../workflow/SideBar/StepSettings';
 import type { IForm } from '../formTypes';
 import { LackIntegrationAlert } from '../LackIntegrationAlert';
-import { VariableManager } from '../VariableManager';
-import { TranslateProductLead } from '../TranslateProductLead';
+import { CustomCodeEditor } from '../CustomCodeEditor';
+import { ChatPreview } from '../../../../components/workflow/Preview';
+
+import { VariableManagerModal } from '../VariableManagerModal';
+import { VariableManagementButton } from '../VariableManagementButton';
 
 const templateFields = ['content'];
 
 export function TemplateChatEditor() {
-  const { readonly } = useEnvController();
+  const [showLoading, setShowLoading] = useState(false);
   const stepFormPath = useStepFormPath();
-  const stepFormErrors = useStepFormErrors();
   const { control } = useFormContext<IForm>();
   const variablesArray = useVariablesManager(templateFields);
   const { hasActiveIntegration } = useHasActiveIntegrations({
     channelType: ChannelTypeEnum.CHAT,
   });
+  const [variablesModalOpened, setVariablesModalOpen] = useState(false);
+
+  const { start, clear } = useTimeout(() => setShowLoading(false), 1000);
+  const handleContentChange = (value: string, onChange: (string) => void) => {
+    setShowLoading(true);
+    clear();
+    onChange(value);
+    start();
+  };
```

Comment:
Suggestion: let's add a comment here why we need the onChange in a timeOut

Reply:
Also, i think we will be using the same logic for other channels too, might be better to abstract this behaviour into a hook. wdyt?

Reply:
Yes, I thought the same, it will be useful. I'll do it